### PR TITLE
Add How-To doc for AWS ECR

### PR DESCRIPTION
**Pull request by KashifSaadat**

### CODE REVIEW

File: how-to-docs/elastic-container-registry.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+1. Login to the Platform Hub via https://hub.acp.homeoffice.gov.uk
+1. Navigate to the Projects list: https://hub.acp.homeoffice.gov.uk/projects/list
+1. Select your Project from the list to go to the detail page (e.g. https://hub.acp.homeoffice.gov.uk/projects/detail/acp)
+    * Ensure you have a **Service** defined within your Project for the Docker Repository to be associated with (check under the `SERVICES` tab)
+1. Select the `ALL DOCKER REPOS` tab
+1. Select the `REQUEST NEW DOCKER REPO` button
+1. Choose the Service to associate this Repository with and provide the name of the Repository to be created (e.g. `hello-world-app`)
+
+The request to create a new Docker Repository can take a few seconds to complete. You can view the status of a Repository by navigating to the `ALL DOCKER REPOS` tab and viewing the list. Once the request has completed, your Repository should have the `Active` label associated with it.
+
```
[the remark this answers is not in the document]
Comment (KashifSaadat) [reply how-to-docs/elastic-container-registry.md]: Yeah that's a good point. I'll shift things around and see how it reads..

File: how-to-docs/elastic-container-registry.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    - docker build -t hello-world-app:$${DRONE_COMMIT_SHA} .
+
+  ecr_push:
+    image: docker:18.09.0
+    secrets:
+    - AWS_ACCESS_KEY_ID
+    - AWS_SECRET_ACCESS_KEY
+    environment:
+    - AWS_DEFAULT_REGION=eu-west-2
+    - AWSCLI_IMAGE=quay.io/ukhomeofficedigital/aws-cli:latest
```
Comment (jaykeshur) [inline how-to-docs/elastic-container-registry.md]: Might be worth doing
`ENV DOCKER_HOST=tcp://172.17.0.1:2375` in the image so we can avoid the line below everywhere

File: how-to-docs/elastic-container-registry.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+* Docker (version 17.06 or above, check with `docker --version`)
+
+### Step 1: Generate IAM Credentials
+
+1. Navigate to the `ALL DOCKER REPOS` tab for your Project within the Platform Hub
+1. For the Repository you have created, select the `MANAGE ACCESS` button
+1. At this stage, you can:
+    * Create a Robot Account, which can be used in deployment pipelines in Drone CI for publishing new images to AWS ECR
+    * Select which Project Members have the ability to pull images, and additionally push updates using their own IAM credentials (separate to the Robot Account and CI builds)
+1. In this example, select your own User, grant write access and press `Save`
```
Comment (jaykeshur) [inline how-to-docs/elastic-container-registry.md]: Think we should avoid putting this in, and state that only the robot user should  have write ideally.

File: how-to-docs/elastic-container-registry.md (hunk 2 above)
Comment (KashifSaadat) [reply how-to-docs/elastic-container-registry.md]: It's to do the `docker run` so would actually need to be in the base dind image (`image: docker:18.09.0`). Might be worth us creating one for use, will simplify builds for users a little bit.

File: how-to-docs/elastic-container-registry.md (hunk 3 above)
Comment (KashifSaadat) [reply how-to-docs/elastic-container-registry.md]: Good point, I'll add a note. I've included here just to help the guide and users testing ECR for the first time.

File: how-to-docs/elastic-container-registry.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+1. Login to the Platform Hub via https://hub.acp.homeoffice.gov.uk
+1. Navigate to the Projects list: https://hub.acp.homeoffice.gov.uk/projects/list
+1. Select your Project from the list to go to the detail page (e.g. https://hub.acp.homeoffice.gov.uk/projects/detail/acp)
+    * Ensure you have a **Service** defined within your Project for the Docker Repository to be associated with (check under the `SERVICES` tab)
+1. Select the `ALL DOCKER REPOS` tab
+1. Select the `REQUEST NEW DOCKER REPO` button
+1. Choose the Service to associate this Repository with and provide the name of the Repository to be created (e.g. `hello-world-app`)
+
+The request to create a new Docker Repository can take a few seconds to complete. You can view the status of a Repository by navigating to the `ALL DOCKER REPOS` tab and viewing the list. Once the request has completed, your Repository should have the `Active` label associated with it.
```
[the remark this answers is not in the document]
Comment (KashifSaadat) [reply how-to-docs/elastic-container-registry.md]: Thanks, added a note.

File: how-to-docs/elastic-container-registry.md (hunk 3 above)
Comment (KashifSaadat) [reply how-to-docs/elastic-container-registry.md]: I've dropped the `grant write access` and added a note, thanks 👍

File: how-to-docs/elastic-container-registry.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+* AWS CLI (version 1.11.91 or above, check with `aws --version`)
+  * Install Guides: [Linux](https://docs.aws.amazon.com/cli/latest/userguide/install-linux.html), [OSX](https://docs.aws.amazon.com/cli/latest/userguide/install-macos.html), [Windows](https://docs.aws.amazon.com/cli/latest/userguide/install-windows.html)
+* Docker (version 17.06 or above, check with `docker --version`)
+
+### Step 1: Generate IAM Credentials
+
+1. Navigate to the `ALL DOCKER REPOS` tab for your Project within the Platform Hub
+1. For the Repository you have created, select the `MANAGE ACCESS` button
+1. At this stage, you can:
+    * Create a Robot Account, which can be used in deployment pipelines in Drone CI for publishing new images to AWS ECR
```
[the remark this answers is not in the document]
Comment (KashifSaadat) [reply how-to-docs/elastic-container-registry.md]: Thanks, updated.

**Pull request merged**